pymeasure/adapters/ni_gpib_232ct.py

```diff
@@ -0,0 +1,346 @@
+        super().__init__(
+            resource_name,
+            asrl={
+                "timeout": serial_timeout,
+                "write_termination": "\r",
+                "read_termination": "\r\n",
+                "flow_control": 2,
+                "chunk_size": 256,
+            },
+            **kwargs,
+        )
+        self.address = address
+        self.connection.write("EOS D")
+        self.connection.write("STAT")
+        self._check_errors()
+
+        if not isinstance(resource_name, NI_GPIB_232):
+            self.eoi = eoi
+        self.flush_read_buffer()
+
+    class GPIBStatus(IntFlag):
+        """Enum element for GIBP status bit decoding
+
+        """
+        ERR = 32768  # Error detected
+        TIMO = 16384  # Time out
+        END = 8192  # EOI or EOS detected
+        SRQI = 4096  # SRQ detected while CIC
+        # 2048, 1024,  & 512 are reserved
+        CMPL = 256  # Operation completed
+        LOK = 128  # Lockout state
+        REM = 64  # remote status
+        CIC = 32  # CIC (Controller in Charge) status
+        ATN = 16  # ATN asserted
+        TACS = 8  # Talker active
+        LACS = 4  # Listener active
+        DTAS = 2  # Device triggeer active state
+        SCAS = 1  # Device Clear active status
+
+    class GPIBError(Flag):
+        """Enum element for GIBP error bit decoding
+
+        """
+        ECMD = 17  # unregcognized command
+        # 15-16 servered
+        EBUS = 14  # Command bytes could not be sent
+        # 12-13 servered
+        ECAP = 11  # No capability for operation
+        # 7-10 reserved
+        EABO = 6  # IO aborted
+        ESAC = 5  # Command requires GPIB-232CT-A to be system controller
+        EARG = 4  # invaild argument(s)
+        EADR = 3  # GPIBN-232CT-A not adressed correctly
+        ENOL = 2  # Write detected, no listeners
+        ECIC = 1  # Command requires GPIB-232CT-A to be CIC
+        NGER = 0  # No error condition
+
+    class SERIALError(Flag):
+        """Enum element for serial error bit decoding
+
+        """
+        EFRM = 4  # Serial port framing error
+        EOFL = 3  # Serial port receive buffer overflow
+        EORN = 2  # Serial port overrrun error
+        EPAR = 1  # Serial port parity error
+        NSER = 0  # No error condition
+
+    def _check_errors(self):
+        """
+        Method to decode the status data reported from the device.
+
+        """
+        self.connection.write("stat n")
+        if self.connection.bytes_in_buffer == 0:
+            time.sleep(0.125)
+        ret_val = self.connection.read_bytes(self.connection.bytes_in_buffer)
+        log.debug(f"read: {ret_val}")
+        if len(ret_val) <= 3:
+            log.warning(f"only {len(ret_val)} bytes received, content: {ret_val}")
+            return
+        if len(ret_val) < 12:
+            log.warning(f"only {len(ret_val)} bytes received, content: {ret_val}")
```

Comment:
Why are 3 or 12 bytes (variying degrees of) bad?

Reply:
The "stat n" command to the GPIB-232CT-A returns normally 4 numeric value representing teh GPIB status, the GPIB error detail, the serial error detail and the count of bytes trensferred by the read/write command before.
These two Ifs were from my debugging sessions with serial comm interruptions.
Will be reomoved